src/nvim/getchar.c

```diff
@@ -263,18 +265,19 @@ add_buff (
         STRLEN(buf->bh_first.b_next->b_str + buf->bh_index) + 1);
   buf->bh_index = 0;
 
-  ssize_t len;
-  if (buf->bh_space >= (int)slen) {
+  size_t len;
+  if (buf->bh_space >= (size_t)slen) {
     len = STRLEN(buf->bh_curr->b_str);
     STRLCPY(buf->bh_curr->b_str + len, s, slen + 1);
-    buf->bh_space -= slen;
+    buf->bh_space -= (size_t)slen;
   } else {
-    if (slen < MINIMAL_SIZE)
+    if (slen < MINIMAL_SIZE) {
       len = MINIMAL_SIZE;
-    else
-      len = slen;
+    } else {
+      len = (size_t)slen;
+    }
```

Comment:
Unrelated, but this entire if block could be `len = MAX(MINIMAL_SIZE, (size_t)slen);`.